tensorflow/compiler/tf2tensorrt/utils/funcdef_to_graphdef.cc

```diff
@@ -0,0 +1,168 @@
+    ndef->set_op(n->type_string());
+    for (const auto& attr : n->attrs()) {
+      (*ndef->mutable_attr())[attr.first] = attr.second;
+    }
+
+    if (!n->assigned_device_name().empty()) {
+      ndef->set_device(n->assigned_device_name());
+    } else {
+      ndef->set_device(n->requested_device());
+    }
+
+    inputs.clear();
+    inputs.resize(n->num_inputs());
+    for (const Edge* e : n->in_edges()) {
+      if (e->IsControlEdge()) {
+        inputs.push_back(e);
+      } else {
+        if (inputs[e->dst_input()] == nullptr) {
+          inputs[e->dst_input()] = e;
+        } else {
+          LOG(WARNING) << "Malformed graph node. multiple input edges: "
+                       << n->DebugString();
+        }
+      }
+    }
+    // node->name() is merely NodeDef::name, which are not guaranteed
+    // to be unique and stable after optimization rewrites. Therefore,
+    // we use "n<node id> or <io prefix><node_id>" instead.
+    for (const Edge* e : inputs) {
+      if (e == nullptr) {
+        ndef->add_input("unknown");
+        continue;
+      }
+      const string srcname = NewNameWithIOPrefix(e->src());
+      if (!e->src()->IsOp()) {
+      } else if (e->IsControlEdge()) {
+        ndef->add_input(strings::StrCat("^", srcname));
+      } else if (e->src_output() == 0) {
+        ndef->add_input(srcname);
+      } else {
+        ndef->add_input(strings::StrCat(srcname, ":", e->src_output()));
+      }
+    }
+  });
+}
+
+Status FunctionDefToGraphDef(FunctionLibraryRuntime::Handle handle,
+                             FunctionLibraryRuntime* flib_runtime,
+                             GraphDef* graph_def, 
+                             std::vector<int>* input_node_ids,
+                             std::vector<int>* output_node_ids) {
+  const FunctionLibraryDefinition* flib_def = flib_runtime->GetFunctionLibraryDefinition();
+  const FunctionBody* fbody;
+  fbody = flib_runtime->GetFunctionBody(handle);
+  //TF_RET_CHECK(*fbody)
+  std::unique_ptr<Graph> graph(new Graph(flib_def));
+    
+  CopyGraph(*fbody->graph, graph.get());
+
+  // Copied from compiler/xla/compile_xla.cc : 
+  /*
+  OptimizerOptions opts;
+  opts.set_opt_level(OptimizerOptions::L0);
+  opts.set_do_common_subexpression_elimination(false);
+  opts.set_do_function_inlining(true);
+  opts.set_do_constant_folding(true);
+  GraphOptimizer optimizer(opts);
+  auto cf_consider_fn = [](const Node* n) {
+    for (const auto& output_arg : n->op_def().output_arg()) {
+      if (output_arg.type() == DT_VARIANT) {
+        return false;
+      }
+    }
+    return true;
+  };
+  GraphOptimizer::Options graph_optimizer_options;
+  graph_optimizer_options.cf_consider_fn = cf_consider_fn;
+  
+  */
+  //optimizer.Optimize(flib_runtime, flib_runtime->env(),
+  //                   /*device=*/nullptr, &graph, graph_optimizer_options);
+   
+  for (Node* n : graph->nodes()) {
+    auto id = n->id();
+    if (n->IsArg()) {
+      VLOG(1) << "Arg Node id " << id;
```

Comment:
Could you make the logs a bit more descriptive about what it's actually doing? But not too long.